[bitnami/external-dns] Add support for NS1 environment variable api key

Laakso opened the pull request:
Signed-off-by: Vesa Laakso <vesa.laakso@rovio.com>

Description of the change

This PR adds the support for NS1 environment variable api key.

Benefits

You can authenticate with Api key.

Possible drawbacks

None.

**Checklist**
<!-- [Place an '[X]' (no spaces) in all applicable fields. Please remove unrelated fields.] -->
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the values.yaml and added to the README.md using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

FraPazGal reviewed (changes requested):
Thanks for the PR @Laakso, could you please check my comments?

Laakso commented on bitnami/external-dns/templates/_helpers.tpl:
```diff
@@ -663,6 +668,19 @@ external-dns: linode.apiToken
 {{- end -}}
 {{- end -}}
 
+{{/*
+Validate values of External DNS:
+- must provide the NS1 API key when provider is "ns1"
+*/}}
+{{- define "external-dns.validateValues.ns1.apiKey" -}}
+{{- if and (eq .Values.provider "ns1") (not .Values.ns1.apiKey) (not .Values.ns1.secretName) -}}
+external-dns: ns1.apiKey
+    You must provide the a NS API key when provider="ns1".
+    Please set the token parameter (--set ns1.token="xxxx")
+    or specify a secret that contains an API token. (--set ns1.secretName="xxxx")
+{{- end -}}
+{{- end -}}
```
I added this validation because api key is required value to set. For some reason it was missing from this helm chart.

https://github.com/kubernetes-sigs/external-dns/blob/master/docs/tutorials/ns1.md

Laakso commented on bitnami/external-dns/templates/_helpers.tpl:
```diff
@@ -663,6 +668,19 @@ external-dns: linode.apiToken
 {{- end -}}
 {{- end -}}
 
+{{/*
+Validate values of External DNS:
+- must provide the NS1 API key when provider is "ns1"
+*/}}
+{{- define "external-dns.validateValues.ns1.apiKey" -}}
+{{- if and (eq .Values.provider "ns1") (not .Values.ns1.apiKey) (not .Values.ns1.secretName) -}}
+external-dns: ns1.apiKey
+    You must provide the a NS API key when provider="ns1".
+    Please set the token parameter (--set ns1.apiKey="xxxx")
+    or specify a secret that contains an API key. (--set ns1.secretName="xxxx")
+{{- end -}}
+{{- end -}}
```
I added this validation because api key is required value to set. For some reason it was missing from this helm chart.

https://github.com/kubernetes-sigs/external-dns/blob/master/docs/tutorials/ns1.md

Laakso commented:
@FraPazGal I added some fixes and checks still pass. : - )

FraPazGal commented on bitnami/external-dns/templates/_helpers.tpl:
```diff
@@ -663,6 +668,19 @@ external-dns: linode.apiToken
+{{- define "external-dns.validateValues.ns1.apiKey" -}}
+{{- if and (eq .Values.provider "ns1") (not .Values.ns1.apiKey) (not .Values.ns1.secretName) -}}
+external-dns: ns1.apiKey
+    You must provide the a NS API key when provider="ns1".
```
```suggestion
    You must provide the NS1 API key when provider="ns1".
```

FraPazGal reviewed (commented):
Nice catch @Laakso! 

One last minor suggestion before merging this. Also, please check the current conflicts.

Laakso commented:
> Nice catch @Laakso!
> 
> One last minor suggestion before merging this. Also, please check the current conflicts.

Removed the extra "a" and fixed merge conflict. : - )

FraPazGal reviewed (approved):
LGTM
